src/zope/event/classhandler.py

```diff
@@ -45,7 +45,10 @@
 registry = {}
 
 def handler(event_class, handler_=None, decorator=False):
-    """Define an event handler for a (new-style) class.
+    """
+    handler(event_class, [handler]) -> None
+
+    Define an event handler for a (new-style) class.
```

Comment:
I'm not a fan of this change.  Why was it made?

Reply:
Because without the signature line you get the real signature, which has an internal-use-only parameter and a parameter that's named-mangled.

Reply:
Agreed:  please leave the imperative mode summary line in place.

Document parameters and return type *below* the summary (see [PEP 257](https://www.python.org/dev/peps/pep-0257/#id17)).

Reply:
Can't you override the signature in Sphinx?  I do that all the time in objgraph.

Also maybe we should rename the `decorator` argument to `_decorator` to indicate it's internal-use-only.

Also also why is the `handler` argument named `handler_`?  It's not clashing with a keyword.

Reply:
This *is* [overriding the signature in Sphinx](http://www.sphinx-doc.org/en/master/ext/autodoc.html#confval-autodoc_docstring_signature):

> autodoc will look at the first line of the docstring for functions and methods, and if it looks like a signature, use the line as the signature and remove it from the docstring content.

The only other way I know to do it is to explicitly write out `..autofunction::` in the .rst file, but that gets back to the repetition I was trying to avoid 😢 

> Also also why is the handler argument named handler_?

Because `handler` is also the name of this function, and this function calls itself recursively.

Reply:
> autodoc will look at the first line of the docstring for functions and methods, and if it looks like a signature, use the line as the signature and remove it from the docstring content.

Nice!  This changes my -0.5 to a -0.

> The only other way I know to do it is to explicitly write out ..autofunction:: in the .rst file, but that gets back to the repetition I was trying to avoid ￼

I was hoping that you could `.. autofunction` one function and have all the others picked up automatically, but I don't know if sphinx supports that, and whether that messes up with the ordering.

> Because handler is also the name of this function, and this function calls itself recursively.

D'oh!